src/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,162 @@
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+/**
+ * The XSLTInjection plugin identifies XSLT injection vulnerabilities with all the parameters in GET and POST
+ *
+ * @author CaptainFreak
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+		
+	private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+	private static final String[] errorCausingPayloads={"<",">","\'","\""};
+	
+	private static final String[] vendorReturningPayloads={"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+															"system-property(\'xsl:vendor\')/>",
+															"\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+															"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+															};
+	private static final String[] xsltVendors={"libxslt","Microsoft","Saxonica","Apache","Xalan","SAXON","Transformiix"};
+	
+	//TODO Add time based payloads (SSRF) and improve accuracy
+	
+	@Override
+	public int getId() {
+		return 90017;
+	}
+
+	@Override
+	public String getName() {		
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.name");
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {  
+		return technologies.includes(Tech.XML); 
+	}
+	
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.refs");
+	}
+
+	@Override
+	public void init() {
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Initialising XSLT Scanner.");
+		}
+	}
+	
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+		try {
+			//Count positive Tests
+			boolean positiveBasedOnError=false; 
+			//error causing tests
+			for (String payload : errorCausingPayloads){
+				msg = getNewMsg();
+				setParameter(msg, param, payload);
+				sendAndReceive(msg);
+				if(!msg.getResponseBody().toString().contains(payload)){
+					positiveBasedOnError=true;
+					break;
+				}
+				// Stop the testing if user requests
+				if (isStop()) {
+					if (LOG.isDebugEnabled()){
+						LOG.debug("Stopping the scan due to a user request");
+					}
+					return;
+				}
+			}
+			//vendor returning tests
+			if(positiveBasedOnError==true){
+				for(String payload : vendorReturningPayloads){
+					msg=getNewMsg();
+					setParameter(msg,param,payload);
+					sendAndReceive(msg);
+					for(String vendor : xsltVendors){
+						if(msg.getResponseBody().toString().contains(vendor)){						
```

Comment:
There seems to be a bunch of trailing tabs here that can be removed.

Reply:
The original response should be pre-checked for these or it may lead to false positives.

Reply:
Like these: https://github.com/zaproxy/zap-extensions/pulls?utf8=%E2%9C%93&q=is%3Apr+pre-check+is%3Aclosed